taskvine/src/manager/vine_file_replica_table.c

```diff
@@ -135,18 +138,16 @@ struct vine_worker_info **vine_file_replica_table_find_replication_targets(
 		if (!peer->transfer_port_active)
 			continue;
 
-		char *peer_addr = string_format("worker://%s:%d", peer->transfer_addr, peer->transfer_port);
 		if (!(replica = hash_table_lookup(peer->current_files, cachename)) &&
 				(strcmp(w->hostname, peer->hostname))) {
-			if ((vine_current_transfers_worker_in_use(q, peer_addr) < q->worker_source_max_transfers) &&
+			if ((vine_current_transfers_source_in_use(q, peer) < q->worker_source_max_transfers) &&
```

Comment:
shouldn't one of these refer to the source, and the other to the destination?

Reply:
Shouldn't source_in_use use worker instead of peer?

In the previous code I was getting tons of transient errors with destinations not able to contact sources. If instead of peer_addr I use worker_addr, then those errors went away. I think that the sources were being overwhelmed.

Reply:
yeah you are right. We should consider worker for dest_in_use(). I think that was an existing bug